### PR TITLE
Fix Push notifications

### PR DESCRIPTION
**Pull request by brunobar79**
- Bump `react-native-firebase`
- Bump `react-native-permissions`
- Got rid of `react-native-push-notifications-ios`
- Fix missing API Key

- Got rid of all these errors:

Screen Shot 2021-07-13 at 6 09 40 PM

DEMO showing WC notifications working on foreground, background and quit states:

iOS:

https://user-images.githubusercontent.com/1247834/125559100-e15f78f1-26fd-4b7b-9617-0fbe3c443944.mov

Android:

https://recordit.co/gQkmmsX4ZH

### CODE REVIEW

**Review by mikedemarais — APPROVED** [review]
LGTM send it but actually

File: src/App.js [hunk 1]
```diff
@@ -216,10 +215,6 @@ class App extends Component {
   };
 
   handleAppStateChange = async nextAppState => {
-    if (nextAppState === 'active') {
```
Comment (jinchung) [inline src/App.js:219]: We don't want this anymore?

File: src/hooks/useScanner.js [hunk 2]
```diff
@@ -39,12 +39,16 @@ function useScannerState(enabled) {
...
-      });
+      request(PERMISSIONS.IOS.CAMERA)
+        .then(permission => {
+          const result = permission === 'granted';
+          if (isCameraAuthorized !== result) {
+            setIsCameraAuthorized(result);
+          }
+        })
+        .catch(e => {
+          logger.log('ERROR REQUESTING CAM PERMISSION', e);
```
Comment (jinchung) [inline src/hooks/useScanner.js:50]: did we want to sentry capture this exception or nah?

File: src/model/firebase.js [hunk 3]
```diff
@@ -18,32 +19,49 @@ export const getFCMToken = async () => {
...
+      permissionStatus === messaging.AuthorizationStatus.PROVISIONAL
+    ) {
+      const fcmToken = await messaging().getToken();
+      if (fcmToken) {
+        saveLocal('rainbowFcmToken', { data: fcmToken });
+      }
     }
   } catch (error) {
-    logger.log('error getting fcm token - cannot save', error);
+    logger.log('error saving fcm token - cannot save', error);
```
Comment (jinchung) [inline src/model/firebase.js:33]: just realized that this error msg should actually be about failing to get the fcm token / permission status, not necessarily about saving it

**Review by jinchung — APPROVED** [review]
🌮 just had a question about removing the other notifications once the user comes back to the app

**Review by skylarbarrera — APPROVED** [review]
😤😤😤

File: src/App.js (hunk 1 above)
Comment (brunobar79) [reply src/App.js:219]: Good catch. Just added it back via https://github.com/rainbow-me/rainbow/pull/2189/commits/4d8efa4a67c9618ea928f81e7642a695ced3bc02

File: src/hooks/useScanner.js (hunk 2 above)
Comment (brunobar79) [reply src/hooks/useScanner.js:50]: That's gonna create a lot of noise from the pro privacy gang

**Review by jinchung — APPROVED** [review]
🍟

**Pull request merged**